fastlane/Fastfile

```diff
@@ -185,22 +185,33 @@ end
 
 desc "Pushes the StreamChat and StreamChatUI SDK podspecs to Cocoapods trunk"
 lane :push_pods do |options|
-  # First pod release will not have any problems
-  pod_push(path: "StreamChat.podspec", allow_warnings: true)
 
-  def release_ui(sync)
+  def release_pod(sync:, podspec:, dry:)
     begin
-      pod_push(path: "StreamChatUI.podspec", allow_warnings: true, synchronous: sync)
-    rescue
-      puts "pod_push failed. Waiting a minute until retry for trunk to get updated..."
+      UI.message "Starting to push podspec: #{podspec}"
+
+      if dry == false
+        pod_push(path: podspec, allow_warnings: true, synchronous: sync)
+      end
+    rescue => exception
+      UI.message exception
+      UI.message "pod_push failed for #{podspec}. Waiting a minute until retry for trunk to get updated..."
       sleep(60) # sleep for a minute, wait until trunk gets updates
-      release_ui(sync)
+      release_pod(sync: sync, podspec: podspec, dry: dry)
     end
   end
 
-  puts "Sleeping for 2 minutes for trunk to get updated..."
-  sleep(60 * 2)
-  release_ui(options[:sync])
+  sync = options[:sync]
+  dry = options[:dry] == 1
```

Comment:
the truthy value conversation we do for this param should be done in this func IMHO. we should also validate the params so we print meaningful errors when users did something incorrectly, like:
> please pass `sync:1` exclude to run in async mode.
> please pass `dry:1` or exclude to disable dry mode.

Reply:
Done! Please re-review @b-onc and let me know your thoughts!